### PR TITLE
Add help intro

### PR DESCRIPTION
**Pull request by bitjammer**
> Note: This PR is for a pitch discussion and building a test toolchain. Please don't merge this – thanks!

Add an "intro" topic to swift-help, printing the version, welcome message, and
available subcommands.

When running `swift` by itself, show the intro before entering the REPL.

When running `swift-help` directly or via `swift --help`, show all of the
available flags as previously.

rdar://78280423

More context: https://forums.swift.org/t/command-line-ux-enhancements-for-swift/50670

### CODE REVIEW

**Comment by bitjammer** [comment]
Please test with following pull request:
https://github.com/apple/llvm-project/pull/3125

@swift-ci Please Build Toolchain macOS Platform

**Comment by bitjammer** [comment]
Please test with the following pull request:
https://github.com/apple/llvm-project/pull/3125

@swift-ci Please Build Toolchain Linux Platform

**Comment by artemcm** [comment]
> Please test with following pull request:
> [apple/llvm-project#3125](https://github.com/apple/llvm-project/pull/3125)
> 
> @swift-ci Please Build Toolchain macOS Platform

@bitjammer , I believe that currently `@swift-ci Please Build Toolchain` only works from PRs on the main [Swift repo](https://github.com/apple/swift).

**Comment by bitjammer** [comment]
I see. @shahmishal @artemcm is it possible to kick off a build manually with the infrastructure referencing these two PRs?

**Comment by artemcm** [comment]
> I see. @shahmishal @artemcm is it possible to kick off a build manually with the infrastructure referencing these two PRs?

You can create an empty PR in the Swift repo and launch a toolchain build referencing both:
apple/swift-driver#772
apple/llvm-project#3125

**Comment by bitjammer** [comment]
Okay I think I will probably hold off on that for the moment, since that could be confusing to the folks that might want it.

**Comment by bitjammer** [comment]
Thank you!

File: Sources/swift-help/main.swift [hunk 1]
```diff
@@ -65,19 +70,60 @@ struct SwiftHelp: ParsableCommand {
         help: "List hidden (unsupported) options")
   var showHidden: Bool = false
 
+  enum Color256: CustomStringConvertible {
```
Comment (bitjammer) [inline Sources/swift-help/main.swift:73]: Wasn't sure where this should go. I think it's somewhat likely that we'll end up using more colors in the future, where we could provide an API that gives the closest color for a given RGB value, but this is only used for one color at the moment, so I placed it here. Let me know if you want this to have a more constrained access level.

**Comment by bitjammer** [comment]
@artemcm Do you know who could review the LLDB change for this? https://github.com/apple/llvm-project/pull/3125

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@artemcm Aside from the conflict, does this look okay to you?

**Comment by bitjammer** [comment]
@swift-ci Please test

File: Sources/SwiftDriver/Driver/CompilerMode.swift [hunk 2]
```diff
@@ -32,6 +32,9 @@
 
   /// Dump information about a precompiled Clang module
   case dumpPCM
+
+  /// Introduce the user to Swift concepts depending on context.
+  case intro
```
Comment (artemcm) [inline Sources/SwiftDriver/Driver/CompilerMode.swift:37]: It seems that this PR does not actually make use of this. I am guessing the intent is to have this become the interactive/contextual driver invocation kind in the future when `swift` no longer launches the REPL?

Comment (artemcm) [reply Sources/SwiftDriver/Driver/CompilerMode.swift:37]: Perhaps this calls us to rename this enum into `DriverMode`, if we're introducing modes of operation to it that may have little to do with the compiler.

**Review by artemcm — APPROVED** [review]
These changes look good to me, and I particularly like the REPL intro. Thanks, @bitjammer.

File: Utilities/build-script-helper.py [hunk 3]
```diff
@@ -197,10 +197,15 @@ def handle_invocation(args):
     print('Cleaning ' + args.build_path)
     shutil.rmtree(args.build_path, ignore_errors=True)
   elif args.action == 'test':
+    print("Files in toolchain bin dir '{}'".format(toolchain_bin))
```
[the remark this answers is not in the document]
Comment (bitjammer) [reply Utilities/build-script-helper.py:200]: Ah, nope; removed, thanks!

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
Looks like an unrelated assertion failure, will try again in a little while.

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by artemcm** [comment]
@swift-ci please test

**Pull request merged**